services/server/server.go

```diff
@@ -546,22 +546,35 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 		return nil, err
 	}
 
-	if err := s.supervisord.UpdateConfiguration(settings); err != nil {
-		return nil, err
-	}
 	if isAgentsStateUpdateNeeded(req.MetricsResolutions) {
 		if err := s.r.UpdateAgentsState(ctx); err != nil {
 			return nil, err
 		}
 	}
-	s.vmdb.RequestConfigurationUpdate()
-	s.vmalert.RequestConfigurationUpdate()
+
+	if err := s.UpdateConfigurations(); err != nil {
```

Comment:
We should move before `s.r.UpdateAgentsState`. Otherwise, agents state can be updated, but then UpdateConfigurations can fail (in theory)